graylog2-web-interface/src/views/components/DashboardSearchBarForm.tsx

```diff
@@ -15,74 +15,66 @@
-import validateTimeRange from 'views/components/TimeRangeValidation';
-import FormWarningsProvider from 'contexts/FormWarningsProvider';
+import FormWarningsContext from 'contexts/FormWarningsContext';
+import type { QueryValidationState } from 'views/components/searchbar/queryvalidation/types';
+import validate from 'views/components/searchbar/validate';
 
-import DateTimeProvider from './searchbar/date-time-picker/DateTimeProvider';
 import { onInitializingTimerange, onSubmittingTimerange } from './TimerangeForForm';
+import DateTimeProvider from './searchbar/date-time-picker/DateTimeProvider';
 
-type Values = {
+export type DashboardFormValues = {
   timerange: TimeRange | undefined | null,
   queryString: string | undefined | null,
 };
 
 type Props = {
-  initialValues: Values,
+  initialValues: DashboardFormValues,
   limitDuration: number,
-  onSubmit: (Values) => void | Promise<any>,
-  children: ((props: FormikProps<Values>) => React.ReactElement) | React.ReactElement,
+  onSubmit: (values: DashboardFormValues) => void | Promise<any>,
+  children: ((props: FormikProps<DashboardFormValues>) => React.ReactElement) | React.ReactElement,
+  validateQueryString: (values: DashboardFormValues) => Promise<QueryValidationState>,
 };
 
-const _isFunction = (children: Props['children']): children is (props: FormikProps<Values>) => React.ReactElement => isFunction(children);
-
-const validate = (nextTimeRange, limitDuration) => {
-  let errors = {};
-
-  const timeRangeErrors = validateTimeRange(nextTimeRange, limitDuration);
+const _isFunction = (children: Props['children']): children is (props: FormikProps<DashboardFormValues>) => React.ReactElement => isFunction(children);
 
-  if (!isEmpty(timeRangeErrors)) {
-    errors = { ...errors, timerange: timeRangeErrors };
-  }
-
-  return errors;
-};
-
-const DashboardSearchForm = ({ initialValues, limitDuration, onSubmit, children }: Props) => {
-  const _onSubmit = useCallback(({ timerange, queryString }) => {
+const DashboardSearchForm = ({ initialValues, limitDuration, onSubmit, validateQueryString, children }: Props) => {
+  const _onSubmit = useCallback(({ timerange, queryString }: DashboardFormValues) => {
     return onSubmit({
       timerange: Object.keys(timerange).length ? onSubmittingTimerange(timerange) : undefined,
       queryString,
     });
   }, [onSubmit]);
   const { timerange, queryString } = initialValues;
-  const initialTimeRange = timerange ? onInitializingTimerange(timerange) : {};
+  const initialTimeRange = timerange ? onInitializingTimerange(timerange) : {} as TimeRange;
   const _initialValues = {
     timerange: initialTimeRange,
     nextTimeRange: initialTimeRange,
     queryString,
   };
 
+  const { setFieldWarning } = useContext(FormWarningsContext);
+  const _validate = useCallback((values: DashboardFormValues) => validate(values, limitDuration, setFieldWarning, validateQueryString),
+    [limitDuration, setFieldWarning, validateQueryString]);
```

Comment:
I am not sure about the reason, but I found the following problem:
When you type in a query like `source:` and you adjust the time range filter, it will:
- validate the query with the old time range (in my example `{}`).
- not update the global override directly, but set the previously configured time range.

Reply:
It looks like the same problem exists with the `SearchBarForm` and the `WidgetQueryControls`.

Reply:
The problem does not occur when changing the streams filter.

Reply:
:white_check_mark: